Sources/Datadog/RUMMonitor.swift

```diff
@@ -245,15 +247,35 @@ public class RUMMonitor: DDRUMMonitor {
         process(
             command: RUMAddUserActionCommand(
                 time: dateProvider.currentDate(),
-                attributes: attributes ?? [:],
+                attributes: aggregate(attributes),
                 actionType: type,
                 name: name
             )
         )
     }
 
+    // MARK: - Attributes
+
+    override public func addAttribute(forKey key: AttributeKey, value: AttributeValue) {
+        queue.async {
+            self.RUMAttributes[key] = value
+        }
+    }
+
+    override public func removeAttribute(forKey key: AttributeKey) {
+        queue.async {
+            self.RUMAttributes[key] = nil
+        }
+    }
+
     // MARK: - Private
 
+    private func aggregate(_ localAttributes: [AttributeKey: AttributeValue]?) -> [AttributeKey: AttributeValue] {
+        var mergedAttributes = queue.sync { return self.RUMAttributes }
```

Comment:
Ref, I created `RUMM-691 Add RUMMonitor benchmarks` so we can test if there's no performance drop due to this extra thread switch for every public API call.

Reply:
i think the cost should be negligible but if we have concerns, we can do that:
```swift
protocol RUMCommand {
  var attributes { get set }
  ...
}
class RUMMonitor {
  func process(cmd) {
    queue.async {
      cmd.attributes.merge(globalAttrs)
      appScope.process(cmd)
    }
  }
```
then we won't need `queue.sync` to get global attrs
let me know if you need this